### DISCUSSION TITLE
kv_cache is only available for cross attention blocks?

### DISCUSSION THREAD

**Discussion by inspirit** [openai/whisper#489] [Q&A] [answered]
Hi!
looking it following line it seems that kv_cache is only enabled for cross_attention blocks?
https://github.com/openai/whisper/blob/9f70a352f9f8630ab3aa0d06af5cb9532bd8c21d/whisper/model.py#L75

**Comment by jongwook** [comment, marked as answer]
Sorry the mechanism is a bit cryptic; the main caching mechanism is implemented in the hooks:

https://github.com/openai/whisper/blob/9f70a352f9f8630ab3aa0d06af5cb9532bd8c21d/whisper/model.py#L250-L255

For the cross-attention, `kv_cache` doesn't have to be updated once computed, whereas the decoder self-attention needs to update the cache in each forward pass by appending the new kv data.

**Reply by inspirit** [reply]
thanks for for a quick reply
